### PR TITLE
research page optimizations

### PR DESCRIPTION
**Pull request by ethn**

### CODE REVIEW

**Comment by ethn** [comment]
I'm not sure it's actually all that much faster yet :(.  But I think the changes are generally positive.

File: mod/wikirate_csv_import/lib/csv_row/structure/answer_csv.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ def check_existence_and_type name, type_id, type_name=nil
         end
       end
 
-      def resolve_source_duplicates existing_source_card
+      def resolve_source_duplication existing_source_card
```
Comment (ethn) [inline mod/wikirate_csv_import/lib/csv_row/structure/answer_csv.rb:69]: please confirm that this change is correct.  `resolve_source_duplicates` is never called. Looked to me like this was intended to be an override.

File: mod/metrics/set/abstract/metric_answer/flags.rb [hunk 2]
```diff
@@ -2,61 +2,54 @@ def imported?
...
-def commented?
-  disc = fetch trait: :discussion
-  disc&.content.present?
-end
-
 format :html do
-  view :flags, cache: :never do
-    output [checked_value_flag, comment_flag, imported_flag, calculated_flag]
+  view :flags do
+    output flags
```
Comment (xithan) [inline mod/metrics/set/abstract/metric_answer/flags.rb]: the `output` is not necessary for arrays in html views

File: mod/metrics/set/abstract/metric_answer/flags.rb [hunk 3]
```diff
@@ -2,61 +2,54 @@ def imported?
...
   end
 
-  view :small_flags, cache: :never do
-    output do
-      [:checked_value, :comment, :imported].map do |flag_name|
-        flag = send "#{flag_name}_flag"
-        "<small>#{flag}</small>"
-      end
+  view :small_flags do
+    output(flags.map { |flag| "<small>#{flag}</small>" })
```
Comment (xithan) [inline mod/metrics/set/abstract/metric_answer/flags.rb]: the same here

File: mod/wikirate_csv_import/lib/csv_row/structure/answer_csv.rb (hunk 1 above)
Comment (xithan) [reply mod/wikirate_csv_import/lib/csv_row/structure/answer_csv.rb:69]: to me too

**Review by xithan — APPROVED** [review]

**Pull request merged**